File: src/Python/281.ZigzagIterator.py

class ZigzagIerator:

    def __init__(self, v1, v2):
        self.vectors = [v1, v2]
        self.row = 0
        self.col = 0
        self.max = max(len(v1), len(v2))

    def next(self): 
        self.hasNext()
        val = self.vectors[self.row][self.col]
        self.row += 1
        return val

    def hasNext(self): 
        while self.col < self.max:
            while self.row < len(self.vectors):
                if self.col < len(self.vectors[self.row]):
                    return True
                self.row += 1
            self.row = 0
            self.col += 1
        return False


if __name__ == "__main__":
    solution = ZigzagIerator([], [])

File: src/CSharp/281.ZigzagIterator.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.ZigzagIterator
{

public class ZigzagIerator {
    List<List<int>> vectors = new List<List<int>>();
    int row, col, max;
    public ZigzagIerator(List<int> v1, List<int> v2) {
        vectors.Add(v1);
        vectors.Add(v2);
        max = Math.Max(v1.Count, v2.Count);
    }

    public int Next() {
        HasNext();
        return vectors[row++][col];
    }

    public bool HasNext() {
        while (col < max) {
            while (row < vectors.Count) {
                if (col < vectors[row].Count)
                    return true;
                row += 1;
            }
            row = 0;
            col += 1;
        }
        return false;
    }
}

}